sdks/python/src/opik/decorator/base_track_decorator.py

```diff
@@ -432,11 +442,25 @@ def _after_call(
                 )
 
             if output is not None:
-                end_arguments = self._end_span_inputs_preprocessor(
-                    output=output,
-                    capture_output=capture_output,
-                    current_span_data=span_data_to_end,
-                )
+                try:
+                    end_arguments = self._end_span_inputs_preprocessor(
+                        output=output,
+                        capture_output=capture_output,
+                        current_span_data=span_data_to_end,
+                    )
+                except Exception as e:
+                    LOGGER.error(
+                        logging_messages.UNEXPECTED_EXCEPTION_ON_SPAN_FINALIZATION_FOR_TRACKED_FUNCTION,
+                        output,
+                        str(e),
+                        exc_info=True,
+                    )
+
+                    error_info = error_info_collector.collect(e)
```

Comment:
I don't think we should collect it, because it's the error that happened in our code, not the user code. And because of that the user will see their spans as failed in the UI (which should be the case if their code fails)